### PR TITLE
Helm chart improvements

### PR DESCRIPTION
**Pull request by blafry**
Hey! I would like to propose minor changes:

- ~extracted registry value to a separate variable (opens up the possibility of using image registry mirrors)~
- dehardcoded controller container ports
- added missing `imagePullSecrets` and `priorityClassName` for default backend
- added missing `readinessProbe` for prometheus-exporter

### CODE REVIEW

**Comment by jcmoraisjr** [comment]
Hi, thanks for the contribution! Can you separate the image registry change into a new PR? My concern is about backward compatibility, so this should be added only in the new v0.15 branch with a proper note in the changelog. Other than that lgtm.

**Comment by blafry** [comment]
@jcmoraisjr Done. https://github.com/haproxy-ingress/charts/pull/61

File: haproxy-ingress/values.yaml [hunk 1]
```diff
@@ -477,6 +482,7 @@ controller:
     # https://github.com/whereisaaron/kube-syslog-sidecar
     # (listens on UDP port 514 and outputs to stdout)
     image:
+      registry: docker.io
```
Comment (jcmoraisjr) [inline haproxy-ingress/values.yaml:485]: This seems not to be in use

**Review by jcmoraisjr — COMMENTED** [review]
Awesome thanks! Just a minor suggestion and we're done.

File: haproxy-ingress/values.yaml (hunk 1 above)
Comment (blafry) [reply haproxy-ingress/values.yaml]: yes, fixed

**Comment by jcmoraisjr** [comment]
Lgtm, thanks! Merging now.

**Pull request merged**